src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Pkcs8PrivateKeyInfo.cs

```diff
@@ -0,0 +1,259 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Security.Cryptography.Asn1;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.Pkcs
+{
+    public sealed class Pkcs8PrivateKeyInfo
+    {
+        public Oid AlgorithmId { get; }
+        public ReadOnlyMemory<byte>? AlgorithmParameters { get; }
+        public CryptographicAttributeObjectCollection Attributes { get; }
+        public ReadOnlyMemory<byte> PrivateKeyBytes { get; }
+
+        public Pkcs8PrivateKeyInfo(
+            Oid algorithmId,
+            ReadOnlyMemory<byte>? algorithmParameters,
+            ReadOnlyMemory<byte> privateKey,
+            bool skipCopies = false)
+        {
+            if (algorithmId == null)
+                throw new ArgumentNullException(nameof(algorithmId));
+
+            AlgorithmId = algorithmId;
+            AlgorithmParameters = skipCopies ? algorithmParameters : algorithmParameters?.ToArray();
+            PrivateKeyBytes = skipCopies ? privateKey : privateKey.ToArray();
+            Attributes = new CryptographicAttributeObjectCollection();
+        }
+
+        private Pkcs8PrivateKeyInfo(
+            Oid algorithmId,
+            ReadOnlyMemory<byte>? algorithmParameters,
+            ReadOnlyMemory<byte> privateKey,
+            CryptographicAttributeObjectCollection attributes)
+        {
+            Debug.Assert(algorithmId != null);
+
+            AlgorithmId = algorithmId;
+            AlgorithmParameters = algorithmParameters;
+            PrivateKeyBytes = privateKey;
+            Attributes = attributes;
+        }
+
+        public static Pkcs8PrivateKeyInfo Create(AsymmetricAlgorithm privateKey)
+        {
+            if (privateKey == null)
+                throw new ArgumentNullException(nameof(privateKey));
+
+            byte[] pkcs8 = privateKey.ExportPkcs8PrivateKey();
+            return Decode(pkcs8, out _, skipCopy: true);
+        }
+
+        public static Pkcs8PrivateKeyInfo Decode(
+            ReadOnlyMemory<byte> source,
+            out int bytesRead,
+            bool skipCopy = false)
```

Comment:
Do we purposefully use `skipCopy` here but `skipCopies` in the public surface area earlier?

Reply:
Different methods.

The ctor takes in two ReadOnlyMemory values, and the same boolean controls both of them.  So it's skipCopies.

Decode takes in one value, so it was skipCopy (singular).  I'm fine with normalizing on plural, rationalized as it being about bytes-plural.